app/models/course/assessment/question/programming.rb

```diff
@@ -110,22 +110,40 @@ def non_autograded_template_files=(template_files)
 
   private
 
+  # Create new package or re-evaluate the old package.
+  def process_package
+    return if @duplicating
+
+    if attachment_changed?
+      return remove_old_package if attachment.nil?
+      process_new_package
+    elsif time_limit_changed? || memory_limit_changed? || language_id_changed?
+      # For non-autograded questions, the attachment is not present
+      evaluate_package if attachment
+    end
+  end
+
+  def evaluate_package
```

Comment:
Should allow `process_new_package` to use `evaluate_package` also? Since there are some commonalities. The `execute_after_commit` is here for some reason also, because we don't want the job to start before the transaction is committed, so we can add that reasoning.